### PR TITLE
Fix for changed API in miekg/dns

### PR DESCRIPTION
**Pull request by tjeb**
PackStruct was removed in miekg/dns; instead, the resolve function now recreates RRs by
writing the RR header information manually into a buffer.

Not sure if this is the best place for it, perhaps something like this should actually be in the exported dns utils (or simply as a function of the RR_Header type), but at least this makes the unbound wrapper work again with the current miekg/dns master branch.

### CODE REVIEW

**Comment by miekg** [comment]
Damn, hadn't realized PackStruct is exported. It should never have been, but maybe we should add it back, but then just calling the correct packer function?

**Comment by tjeb** [comment]
yeh if there was a packheader it could just call that :)

**Comment by miekg** [comment]
Same for UnpackStruct

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [miekg/unbound] Fix for changed..." ]
>yeh if there was a packheader it could just call that :)

There is, but not exported. Let's move the discussion to:
https://github.com/miekg/dns/issues/378

Up for ideas. But bringing back Unpack/pack-Struct is not a bad thing
per se.

/Miek

--
Miek Gieben

File: unbound.go [hunk 1]
```diff
@@ -244,12 +245,23 @@ func (u *Unbound) Resolve(name string, rrtype, rrclass uint16) (*Result, error)
...
+			binary.BigEndian.PutUint16(msg[off:], h.Rrtype)
+			off += 2
+			binary.BigEndian.PutUint16(msg[off:], h.Class)
+			off += 2
+			binary.BigEndian.PutUint32(msg[off:], h.Ttl)
+			off += 4
+			binary.BigEndian.PutUint16(msg[off:], h.Rdlength)
+			off += 2
 			rrbuf := append(msg, b...)
+
```
Comment (miekg) [inline unbound.go]: Just exporting the pack/unpack for the header would fix this. Pondering a better fix. IIRC `res.data` holds the RR data without the header.

Comment (miekg) [inline unbound.go]: Also this header is the same for all RR, why recode in the loop everything time...?

Comment (tjeb) [inline unbound.go]: oh, yes, that could be pulled out of the loop

Comment (miekg) [inline unbound.go]: [ Quoting <notifications@github.com> in "Re: [miekg/unbound] Fix for changed..." ]
>>  			rrbuf := append(msg, b...)
>> +
>
>oh, yes, that could be pulled out of the loop

Well, if you still have this code loaded in your $EDITOR...

Comment (tjeb) [inline unbound.go]: Done, see https://github.com/miekg/unbound/pull/7

**Pull request merged**